File: task/src/main/java/com/goexp/galgame/data/piplline/core/Message.java
package com.goexp.galgame.data.piplline.core;

public class Message<T> {

    public int code;

    public T entity;

    public Message() {
    }

    public Message(int code, T entity) {
        this.code = code;
        this.entity = entity;
    }

    @Override
    public String toString() {
        return "Message{" +
                "code=" + code +
                '}';
    }
}

File: piplline/src/main/scala/com/goexp/piplline/core/Message.scala
package com.goexp.piplline.core

case class Message(code: Int, entity: Any) {

  def this(clazz: Class[_], entity: Any) =
    this(clazz.hashCode(), entity)


  override def toString: String = "Message{" + "code=" + code + '}'
}